### DISCUSSION TITLE
update of extension for typescript

### DISCUSSION THREAD

**Discussion by indolent-developer** [avajs/ava#2989] [General]
I was looking at this code block.

```
if (Reflect.has(conf, 'typescript')) {
		try {
			const {level, main} = await providerManager.typescript(projectDir);
			providers.push({
				level,
				main: main({config: conf.typescript}),
				type: 'typescript',
			});
		} catch (error) {
			exit(error.message);
		}
	}
```

...

`let extensions;
	try {
		extensions = normalizeExtensions(conf.extensions, providers);
	} catch (error) {
		exit(error.message);
	}`

so as per this link. https://github.com/avajs/ava/blob/main/docs/recipes/typescript.md

if compile= false , should we be still added ts in the extension list. The code is adding ts extension, which is unintentionally testing ts files for me.

**Comment by novemberborn** [comment]
Yes, I use this without explicitly compiling. Instead I rely on a separately running `tsc --watch`.

> The code is adding ts extension, which is unintentionally testing ts files for me.

Could you elaborate? Why are you using `@ava/typescript` if you don't want to test `ts` files?
